### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by khomutov-k**
Home work 5 is done.

### CODE REVIEW

File: src/main/java/hw5/page/UserTablePage.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    private WebDriver driver;
+
+    public UserTablePage(WebDriver driver) {
+        PageFactory.initElements(driver,this);
+        this.driver = driver;
+    }
+
+    public String getTitle() {
+        return driver.getTitle();
+    }
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/page/UserTablePage.java]: In genereal these constructor and method could be extracted to the `Base` class

File: src/test/java/hw3/TestCommons.java [hunk 2]
```diff
@@ -1,26 +0,0 @@
-package hw3;
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/TestCommons.java:1]: Why do you delete this files of the `hw3`?

File: src/main/java/hw5/steps/HomePageSteps.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class HomePageSteps {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/steps/HomePageSteps.java]: What is the purpose of the current class?

File: src/test/java/hw5/steps/AssertionSteps.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    @Then("Log row has {string} text in log section")
+    public void logRowHasTextInLogSection(String expectedLog) {
+        SoftAssertions softAssertions = new SoftAssertions();
+        Helper.textOfWebElementsContainsSetText(userTablePage.getLogLines(),
+                new HashSet<>(Collections.singletonList(expectedLog)),
+                softAssertions);
+        softAssertions.assertAll();
+    }
+
+    @And("Description text under image in {string} row should be displayed as {string} with {string}")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/steps/AssertionSteps.java]: Please do not use `@And` and `@But` annotations in the step definitions class. This is not clear what is the purpose to the method

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please resolve comments

File: src/main/java/hw5/steps/HomePageSteps.java (hunk 3 above)
Comment (khomutov-k) [reply src/main/java/hw5/steps/HomePageSteps.java]: It was copied from previous hw. I will remove it to make clear location for actual steps.

File: src/test/java/hw3/TestCommons.java (hunk 2 above)
Comment (khomutov-k) [reply src/test/java/hw3/TestCommons.java:1]: It was unnecessary for this hw, and I removed tests to start from scratch. How should I fix it?)

File: src/main/java/hw5/page/UserTablePage.java (hunk 1 above)
Comment (khomutov-k) [reply src/main/java/hw5/page/UserTablePage.java]: I will make `BasePage` class for this reason.

File: src/test/java/hw5/steps/AssertionSteps.java (hunk 4 above)
Comment (khomutov-k) [reply src/test/java/hw5/steps/AssertionSteps.java]: Ok, thank you for your notes. I will avoid this annotations.

**Review by DmitryKhodakovsky — APPROVED** [review]
Approved. Could be merged to the master branch

**Pull request merged**